d2game/d2player/escape_menu.go

```diff
@@ -414,6 +427,23 @@ func (m *EscapeMenu) showLayout(id layoutID) {
 	}
 
 	if id == saveLayoutID {
+		// Save Game
+		heroState := d2hero.HeroState{
+			HeroName:   m.gameClient.Players[m.gameClient.PlayerID].Name(),
+			HeroType:   m.gameClient.Players[m.gameClient.PlayerID].Class,
+			HeroLevel:  m.gameClient.Players[m.gameClient.PlayerID].Stats.Level,
+			Act:        1,
+			FilePath:   "",
+			Equipment:  *m.gameClient.Players[m.gameClient.PlayerID].Equipment,
+			Stats:      m.gameClient.Players[m.gameClient.PlayerID].Stats,
+			Skills:     m.gameClient.Players[m.gameClient.PlayerID].Skills,
+			X:          m.gameClient.Players[m.gameClient.PlayerID].Position.X(),
+			Y:          m.gameClient.Players[m.gameClient.PlayerID].Position.Y(),
+			LeftSkill:  m.gameClient.Players[m.gameClient.PlayerID].LeftSkill,
+			RightSkill: m.gameClient.Players[m.gameClient.PlayerID].RightSkill,
+		}
+		m.heroState.Save(&heroState)
```

Comment:
I would not expect that the saving of the hero happens in the escape menu. I would rather put that into the player class as you only used properties from that class.  Then you can call that method here.

Reply:
Of this I was aware, and yes, that will be changed too. I already told grave in Discord that I don't find it very good that the esacpe menu saves a character. That doesn't make much sense.
After I got it to work I will also refactor that.

With this WIP: PR I only wanted to check if I'm completely off track here. But it does not seem so. Thanks for your help!